src/components/TransactionBlock/WalletTransactionBlock.tsx

```diff
@@ -0,0 +1,737 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { ethers } from 'ethers';
+import { AccountBalance, AccountTypes } from 'etherspot';
+
+import TextInput from '../TextInput';
+import { useEtherspot, useTransactionBuilder } from '../../hooks';
+import { formatAmountDisplay, formatAssetAmountInput, formatMaxAmount } from '../../utils/common';
+import { Chain, supportedChains } from '../../utils/chain';
+import NetworkAssetSelectInput from '../NetworkAssetSelectInput';
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+import { CombinedRoundedImages, RoundedImage } from '../Image';
+import { Pill } from '../Text';
+import { Theme } from '../../utils/theme';
+import AccountSwitchInput from '../AccountSwitchInput';
+import { IMultiCallData, ITransactionBlock } from '../../types/transactionBlock';
+import {
+  WalletAssetSearchIcon,
+  WalletBridgeIcon,
+  WalletDepositIcon,
+  WalletSendIcon,
+  WalletSwapIcon,
+  WalletWithdrawIcon,
+  WalletDropdownUpIcon,
+  WalletDropdownDownIcon,
+  WalletCopyIcon,
+  WalletCloseSearchIcon,
+} from './Icons';
+import { Text } from '../Text';
+import { TRANSACTION_BLOCK_TYPE, TRANSACTION_BLOCK_TYPE_KEY } from '../../constants/transactionBuilderConstants';
+import { ISendAssetTransactionBlockValues } from './SendAssetTransactionBlock';
+
+interface IChainAssets {
+  title: string;
+  chainId: number;
+  chain: Chain;
```

Comment:
both `chain` and `chainId` defines chain ID, anything missing here for one of chain ID naming or duplicate?
```suggestion
  chainId: number;
  chain: Chain;
```

Reply:
`chain` gives you some other values like title + iconUrl, but you're right i've just removed `chainId`